src/storage/browser.js

```diff
@@ -32,6 +34,8 @@ const BrowserStorageFactory = (settings) => {
             segments: new SegmentCacheInMemory(childKeyBuilder),
             impressions: this.impressions,
             metrics: this.metrics,
+            // @TODO review this because I'm not sure this will work with shared instances
+            count: this.count,
```

Comment:
Can you give me some more context on why you added these TODO? I don't remember why you've left them.